app/src/main/java/com/monkeyteam/chimpagne/ui/FindEventScreen.kt

```diff
@@ -0,0 +1,370 @@
+  val goToMap: () -> Unit = {
+    coroutineScope.launch { pagerState.animateScrollToPage(FindEventScreens.MAP) }
+  }
+
+  ChimpagneTheme {
+    HorizontalPager(state = pagerState, userScrollEnabled = false) { page ->
+      when (page) {
+        FindEventScreens.FORM -> FindEventFormScreen(navObject, goToMap)
+        FindEventScreens.MAP -> FindEventMapScreen(goToForm)
+      }
+    }
+  }
+}
+
+@ExperimentalMaterial3Api
+@Composable
+fun FindEventFormScreen(navObject: NavigationActions, onSearchClick: () -> Unit) {
+
+  var showDatePicker by remember { mutableStateOf(false) }
+  val datePickerState = rememberDatePickerState()
+
+  var selectedDate by remember { mutableStateOf(Calendar.getInstance()) }
+
+  var tagInput by remember { mutableStateOf("") }
+  val selectedTags = remember { mutableStateListOf<String>() }
+
+  var searchRadius by remember { mutableStateOf(1f) }
+
+  Scaffold(
+      topBar = {
+        TopAppBar(
+            title = { Text("Find an event") },
+            modifier = Modifier.shadow(4.dp),
+            navigationIcon = {
+              IconButton(onClick = { navObject.goBack() }) {
+                Icon(Icons.AutoMirrored.Filled.ArrowBack, "back")
+              }
+            })
+      },
+      bottomBar = {
+        Button(
+            onClick = { onSearchClick() },
+            modifier =
+                Modifier.fillMaxWidth().padding(8.dp).height(56.dp), // Typical height for buttons
+            shape = MaterialTheme.shapes.extraLarge) {
+              Icon(Icons.Rounded.Search, contentDescription = "Search")
+              Spacer(Modifier.width(8.dp))
+              Text("Search", style = MaterialTheme.typography.bodyLarge)
+            }
+      }) { innerPadding ->
+        Box(modifier = Modifier.padding(innerPadding)) {
+          Column(
+              modifier =
+                  Modifier.fillMaxSize().padding(16.dp).verticalScroll(rememberScrollState()),
+              horizontalAlignment = Alignment.Start) {
+                FindEventLegend("Event Location", Icons.Rounded.Search, "Search")
+
+                Spacer(Modifier.height(16.dp))
+
+                LocationFinder("", {}, emptyList(), {}, Modifier.fillMaxWidth())
+                Spacer(Modifier.height(16.dp))
+                IconTextButton(
+                    text = "Locate me",
+                    icon = Icons.Rounded.MyLocation,
+                    onClick = { /*TODO*/},
+                    modifier = Modifier.align(Alignment.CenterHorizontally))
+                Spacer(Modifier.height(16.dp))
+
+                Text(text = "Search Radius: ${searchRadius.toInt()} km")
+                Slider(
+                    value = searchRadius,
+                    onValueChange = { searchRadius = it },
+                    valueRange = 1f..30f,
+                    modifier = Modifier.fillMaxWidth())
+
+                Spacer(Modifier.height(40.dp))
+
+                FindEventLegend("Add Tags to your search", Icons.Rounded.Tag, "Tags")
```

Comment:
Use strings defined in resources for text, this allows for translations

Reply:
Done